horovod/common/operations.cc

```diff
@@ -421,9 +421,12 @@ void BackgroundThreadLoop(HorovodGlobalState& state) {
     state.timeline.Initialize(horovod_timeline,
                               static_cast<unsigned int>(size));
   }
-  if (horovod_timeline != "") {
-      should_enable_timeline = true;
+
+  should_enable_timeline = true;
+  if (horovod_timeline == "DISABLED") {
```

Comment:
Is this debug code?

Reply:
I kept it  as emergency switch to completely Disable and short circuit timeline intialization and fallback to earlier behavior.

Reply:
I see, so even if the user does not specify a timeline path, we still enable it in case they want to specify one later on.  I wonder if this is related to a segfault a user reported recently: #2365.

Can you take a look?

Reply:
Yes, we make sure that timeline bits are always there, but timeline.cc would control and return with no-op if timeline is disabled. 
For 2365, stack trace doesn't tell much. though doesn't look like TL related as error is thrown at this point -  https://github.com/horovod/horovod/blob/master/horovod/torch/ready_event.cc#L94